examples/05-file-IO/02-hdf5_serialize_and_read.py

```diff
@@ -0,0 +1,160 @@
+# noqa: D400
+"""
+.. _ref_basic_hdf5:
+
+HDF5 export and import operations
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This example shows you how to use the HDF5 format to export results
+and meshed regions in an H5 file.
+It also demonstrates how to read results and meshed regions from the
+created H5 file.
+
+First, it exports all the results for all time frequencies,
+then it exports all the time sets for the results, per time set.
+Finally, it reads the results and compares them.
+For the example to run correctly, ensure you do not have an existing H5 file.
+
+"""
+
+###############################################################################
+# Import modules, instantiate model and create temporary folder
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+# Import the ``dpf-core`` module and its examples files.
+
+import tempfile
+
+import ansys.dpf.core as dpf
+from ansys.dpf.core import examples
+
+###############################################################################
+# Instantiate the model and the provider operators:
+
+model = dpf.Model(examples.download_transient_result())
+streams_cont = model.metadata.streams_provider.outputs.streams_container
+time_freq_op = dpf.operators.metadata.time_freq_provider(streams_container=streams_cont)
+time_freq_support = time_freq_op.outputs.time_freq_support()
+time_freqs = time_freq_support.time_frequencies
+
+result_names_on_all_time_steps = []
+result_names_time_per_time = []
+
+num_res = len(model.results)
+num_sets = len(time_freqs.data)
+
+###############################################################################
+# Define a temporary folder for outputs:
+if dpf.SERVER.local_server:
```

Comment:
@anslpa please do not use explicit 'tmp' folre anymore.
you can see here how to do it #936